build_tools/ci/cpu_comparison/input_generator.py

```diff
@@ -165,10 +172,45 @@ def write_input(bin_filename, num_elements, element_type, input_number, input_se
         file.write(data)
 
 
-def generate_inputs(filename, write_dir, seed):
+def write_input(bin_filename, num_elements, element_type, np_array):
+    """
+    Write the numpy array `np_array` to the binary file `bin_filename`. The
+    number of elements in `np_array` must be `num_elements` (this is verified).
+    The elements in `np_array` will be cast to the data type `element_type`,
+    and so can be of any type.
+    """
+    # Assert that the number of elements is correct:
+    if num_elements != np_array.size:
+        raise ValueError(
+            f"Expected {num_elements} elements, but got {np_array.size} elements."
+        )
+
+    if element_type == "bf16":
+        array_f32 = np_array.astype(np.float32)
+        data = np.array(
+            [convert_f32_to_bf16(f) for f in array_f32], dtype=np.uint16
```

Comment:
TODO(newling) find nicer way of casting numpy arrays to bf16 (I think this is what @jtuyls was suggesting https://github.com/nod-ai/iree-amd-aie/pull/829#discussion_r1790210012)